gh-96348: Deprecate the 3-arg signature of coroutine.throw, generator.throw and agen.athrow (#96428)
python/cpython: Doc/reference/expressions.rst

   :attr:`~BaseException.__traceback__` attribute stored in *value* may
   be cleared.

>    .. deprecated:: 3.12

Reviewer: I think this would just be 

`.. versionchanged:: 3.12`

>       The second signature \(type\[, value\[, traceback\]\]\) is deprecated now.
>       It is still available, but may be removed in a future version of Python.

Reviewer: ```suggestion
      The second signature \(type\[, value\[, traceback\]\]\) is deprecated and
      may be removed in a future version of Python.
```

Reviewer: There are a few other places in the docs (I used index search for throw):

https://docs.python.org/3.12/reference/datamodel.html?highlight=throw#coroutine.throw
https://docs.python.org/3.12/reference/expressions.html?highlight=throw#agen.athrow

And as @kumaraditya303 mentioned, we need an entry in 

https://github.com/python/cpython/blob/main/Doc/whatsnew/3.12.rst

   raises a different exception, then when the awaitable is run that exception
   propagates to the caller of the awaitable.

   .. versionchanged:: 3.12

      The second signature \(type\[, value\[, traceback\]\]\) is deprecated and

>       may be removed in a future version of Python.

Reviewer: I don't think this PR actually raises a deprecation error for this one.
